pkg/httpapi/api.go

```diff
@@ -161,6 +166,41 @@ func (a *APIRouter) GetApplication(w http.ResponseWriter, r *http.Request) {
 	marshalResponse(w, appEnvironments)
 }
 
+func (a *APIRouter) ListApplications(w http.ResponseWriter, r *http.Request) {
+	repoURL := strings.TrimSpace(r.URL.Query().Get("url"))
```

Comment:
Do we need the URL query param? We can extract it directly from `spec.source.repoURL`

Reply:
So we moved away from the user specif checks and are replicating the behaviour of the existing `/pipelines` API, i.e the check is based on the `gitops-url` provided in the req. 
So this query param is needed to filter out all the applications with `spec.source.repoURL` as the `req url`. 
So Yes, we need this !!